book/custom_commands.md

```diff
@@ -252,187 +444,350 @@ def greet [
+    --caps: bool   # Not allowed
 ] { ... }
 ```
 
-instead, you should define it as a basic switch: `def greet [--twice] { ... }`
+:::
 
-Flags can contain dashes. They can be accessed by replacing the dash with an underscore:
+Flags can contain dashes. They can be accessed by replacing the dash with an underscore in the resulting variable name:
 
 ```nu
 def greet [
   name: string
-  --age (-a): int
-  --two-times
+  --all-caps
 ] {
-  if $two_times {
-    [$name $name $age $age]
-  } else {
-    [$name $age]
-  }
+    let greeting = $"Hello, ($name)!"
+    if $all_caps {
+      $greeting | str upcase
+    } else {
+      $greeting
+    }
 }
 ```
 
-## Rest parameters
+### Rest parameters
 
-There may be cases when you want to define a command which takes any number of positional arguments. We can do this with a rest parameter, using the following `...` syntax:
+There may be cases when you want to define a command which takes any number of positional arguments. We can do this with a "rest" parameter, using the following `...` syntax:
 
 ```nu
-def greet [...name: string] {
-  print "hello all:"
-  for $n in $name {
-    print $n
+def multi-greet [...names: string] {
+  for $name in $names {
+    print $"Hello, ($name)!"
   }
 }
 
-greet earth mars jupiter venus
+multi-greet Elin Lars Erik
+# => Hello, Elin!
+# => Hello, Lars!
+# => Hello, Erik!
 ```
 
 We could call the above definition of the `greet` command with any number of arguments, including none at all. All of the arguments are collected into `$name` as a list.
 
 Rest parameters can be used together with positional parameters:
 
 ```nu
-def greet [vip: string, ...name: string] {
-  print $"hello to our VIP ($vip)"
-  print "and hello to everybody else:"
-  for $n in $name {
-    print $n
+def vip-greet [vip: string, ...names: string] {
+  for $name in $names {
+    print $"Hello, ($name)!"
   }
+
+  print $"And a special welcome to our VIP today, ($vip)!"
 }
 
-#     $vip          $name
-#     ---- ------------------------
-greet moon earth mars jupiter venus
+#         $vip          $name
+#         ----- -------------------------
+vip-greet Rahul Priya Arjun Anjali Vikram
+# => Hello, Priya!
+# => Hello, Arjun!
+# => Hello, Anjali!
+# => Hello, Vikram!
+# => And a special welcome to our VIP today, Rahul!
 ```
 
-To pass a list to a rest parameter, you can use the [spread operator](/book/operators#spread-operator) (`...`):
+To pass a list to a rest parameter, you can use the [spread operator](/book/operators#spread-operator) (`...`). Using the `vip-greet` command definition above:
 
 ```nu
-> let planets = [earth mars jupiter venus] # This is equivalent to the previous example
-> greet moon ...$planets
+let vip = "Tanisha"
+let guests = [ Dwayne, Shanice, Jerome ]
+vip-greet $vip ...$guests
+# => Hello, Dwayne!
+# => Hello, Shanice!
+# => Hello, Jerome!
+# => And a special welcome to our VIP today, Tanisha!
 ```
 
-## Documenting Your Command
+## Input-Output Signature
```

Comment:
Would it be possible to name this "Input-Output Pipeline Signature" or "Pipeline Input-Output Signature"? I know I've tried looking for this section before by searching for "pipeline signature", and it could make it a bit clearer what it's referring to

Reply:
In upcoming commit